simple_database/main.py

```diff
@@ -1,5 +1,219 @@
+        if action == 'connect' and os.path.isfile(self.file_path):
+            self.load()
+        if action == 'create' and os.path.isfile(self.file_path):
+            raise ValidationError('Database with name "{}" already exists.'.format(db_name))
+
+    def save(self):
+        # Little function used in save() to convert the date to and from datetime/str for use in JSON
+        def convert_rows(rows):
+            new_rows = []
+
+            for row in rows:
+                new_row = []
+                for field in row:
+                    # If item is a date object, convert to ordinal so it can be stored in JSON
+                    if isinstance(field, date):
+                        field = field.toordinal()  #To Do: reverse conversion from ordinal to date object on load
+                    new_row.append(field)
+                new_rows.append(new_row)
+
+            return new_rows
+
+        data = {}
+
+        for table in self.tables:
+            table_obj = getattr(self, table)
+            data[table] = {
+                'columns': table_obj.columns,
+                'rows': convert_rows(table_obj.rows),
+            }
+
+        with open(self.file_path, 'w') as f:
+            json.dump(data, f)
+
+    def load(self):
+        with open(self.file_path, 'r') as f:
+            data = json.load(f)
+
+        for k, v in data.items():
+            table = Table(v['columns'], self)
+            table.rows = v['rows']
+            self.tables.append(k)
+            setattr(self, k, table)
+
+    def create_table(self, table_name, columns):
+        # Set table objects as attributes to db
+        setattr(self, table_name, Table(columns, self))
+
+        # Add name to table list
+        self.tables.append(table_name)
+
+        # Save state to file
+        self.save()
+
+    def show_tables(self):
+        return self.tables
+
+class Table(object):
+    def __init__(self, columns, db):
+        self.columns = columns
+        self.rows = []
+        self.db = db
+
+        for field in columns:
+            setattr(self, field.get('name'), field.get('type')) # should type be set like this?
+
+    def insert(self, *args):
+        # Check amount of provided args
+        if len(args) != len(self.columns):
+            raise ValidationError('Invalid amount of field')
+
+        # Check data types
+        i = 0
+        for item in self.columns:
+            if isinstance(args[i], eval(item.get('type'))):
+                pass
+            else:
+                raise ValidationError('Invalid type of field "{}": Given "{}", expected "{}"'.format(item.get('name'), type(args[i]).__name__, item.get('type')))
+                
+            i += 1
+
+        # Create new row (list) in database file
+        self.rows.append([arg for arg in args])
+
+        # Save state to file
+        self.db.save()
+
+    #def query():
+    
+    
+    #def describe():
+
+
+    def all(self):
+        return [row for row in self.rows]
```

Comment:
I wonder if that works for you. I think the tests are calling next on the return value of all(). And I don't think lists have a next.
````python
#Maybe:
return iter([n for n in stuff])
#or a generator
return ( n for n in stuff)
````